### ISSUE TITLE
Configure repository settings

### ISSUE THREAD

**Issue by amotl** [jpmens/mqttwarn#512]
Dear @jpmens,

I would like to update some settings on this repository like branch protection rules and such. Can I humbly ask you for corresponding permissions if you trust me enough?

With kind regards,
Andreas.

**Comment by jpmens** [comment]
It's a single repository, not part of an organization. I can't see how to change permissions for a collaborator, and the documentation doesn't explain so as far as I can tell.

If you tell me which rules we should apply in total I see if we can add those.

**Comment by amotl** [comment]
Hi JP,

all right, thank you. I will just go ahead and list up my requests then?

- Rename "master" branch to "main", as this is also the new default on GitHub.
  There is a "rename branch" action at https://github.com/jpmens/mqttwarn/settings/branches. I will accompany this update by a corresponding patch.
- Adjust branch protection rules.
  If you also like the new CI on GHA, let's uncheck the requirement on CircleCI but check all other individual items. Maybe let's leave all Python 2.x items unchecked because we might drop support for it anytime soon.
- Fix typo.
  The "About project" description »Subscribe to MQTT topics (with wildcards) and notifiy pluggable services« has a typo within notifiy.

With kind regards,
Andreas.

**Comment by jpmens** [comment]
Andreas,

* why should we rename the master branch? I see no reason to do so? It may be the new Github default, but that doesn't mean we need to change, in fact most repositories haven't renamed.
* I'm not sure I agree with several of the "branch protection rules", and I'm possibly not seeing what you see as there's no mention of "python2" at [1].
* I have enabled auto-delete on head branches
* I have fixed the typo, thanks.

**Comment by amotl** [comment]
Hi JP,

Thank you very much.

- Regarding the "main" branch thing: It is not only GitHub which is updating their defaults, it's also GitLab and Git itself [1,2,3] and we started following that movement already on behalf of other repositories. The effort is quite low, see #517.
- The "branch protection rules" thing: I am referring to the list of checks running on the code base as can be inspected at #518, which I submitted for demonstration purposes. Specific checks can be made obligatory to pass successfully, i.e. _required_.

If both things are nothing which resonates with you, we can also leave everything as is and continue chugging along. Feel free to just close this issue then.

With kind regards,
Andreas.

[1] https://lore.kernel.org/git/pull.656.v4.git.1593009996.gitgitgadget@gmail.com/
[2] https://about.gitlab.com/blog/2021/03/10/new-git-default-branch-name/
[3] https://www.theserverside.com/feature/Why-GitHub-renamed-its-master-branch-to-main

**Comment by jpmens** [comment]
You originally wrote:

* Adjust branch protection rules at [1].
  If you also like the new CI on GHA, let's uncheck the requirement on CircleCI but check all other individual items. Maybe let's leave all Python 2.x items unchecked because we might drop support for it anytime soon.

I do not see that anywhere, sorry:

![rabbit-8066](https://user-images.githubusercontent.com/60706/122171886-83f5ec80-ce80-11eb-9098-f37b5c053fcf.png)

![rabbit-8067](https://user-images.githubusercontent.com/60706/122171924-8e17eb00-ce80-11eb-90b2-01e7d45a922c.png)

![rabbit-8068](https://user-images.githubusercontent.com/60706/122171960-95d78f80-ce80-11eb-9c48-52e041e4d609.png)

**Comment by amotl** [comment]
Ah I see. One probably would have to add a new rule first by pressing the "Add rule" button. Then, the list of available status checks will probably appear within this section.

**Comment by jpmens** [comment]
Yes, that's accurate; so you want me to tick all the boxes?

**Comment by amotl** [comment]
First, I just wanted to present it as an idea to you. Second, if you agree, then please spare the item saying s.th. about Python 2.7, this will probably be dropped soon. Please also spare the item about CircleCI.

**Comment by jpmens** [comment]
I've added a rule on `main` which requires pull request reviews before merging; if we need more we can add them as we go.

**Comment by amotl** [comment]
Thank you!

**Comment by amotl** [comment]
Hi @jpmens,

unfortunately, with the current repository settings, I can't issue releases anymore.

```
$ make release bump=minor
git push && git push --tags
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 16 threads
Compressing objects: 100% (9/9), done.
Writing objects: 100% (9/9), 835 bytes | 835.00 KiB/s, done.
Total 9 (delta 7), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (7/7), completed with 6 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/main.
remote: error: At least 1 approving review is required by reviewers with write access.
To github.com:jpmens/mqttwarn.git
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'github.com:jpmens/mqttwarn.git'
make: *** [Makefile:72: push] Error 1
```

May I ask you to relax those constraints again?

With kind regards,
Andreas.

**Comment by jpmens** [comment]
I have deleted the branch protection rule.

**Comment by amotl** [comment]
Thank you! I just released mqttwarn 0.25.0.
